projects/cdk/utils/miscellaneous/provide.ts

```diff
@@ -0,0 +1,18 @@
+import type {ExistingProvider, ProviderToken} from '@angular/core';
+
+export function tuiProvide<T>(
+    provide: ProviderToken<T>,
+    useExisting: ProviderToken<T>,
+): ExistingProvider;
+export function tuiProvide<T>(
+    provide: ProviderToken<T | T[]>,
+    useExisting: ProviderToken<T>,
+    multi: boolean,
+): ExistingProvider;
+export function tuiProvide<T>(
```

Comment:
can you explain please why we need such overload? Can't figure out 🤔  [ts](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAE2AYwDYEMrDgSwHYzBQBm6y2AoiDgM4z4DmAClBAG45JRwDeAUHEFwArjWBVa9PAwBccdHgCeAbgFCwrDkjkKVawQFthqegH45AIwgRUwBaoC+fPkjSZsbmjTgBJPACsUegg8ABUIAGtgPAAeUIA+Xn04DQhCZEIEOAB9JBpkOTooRlUhOCx0BBDURThAVyBUtjlhPAi8CAB3PFVk5BCi4QzoAApc4HzCmGLpABo4CDBgvBpzJLKyxs5gBD9VgHJWNL24AB84PbAMGGJoA2Ozvd17uDxjVFL1wVIhqEU5YYAlHAALyJUIfQQOAEQuAwCAAZSmjEBk2mDEczhgijA2BY7C2UHCUViCRBvgCQRwISJ0Ti8WcfFAkFgcGILQyVLwsOEODxWmAdOGyU22jgfIJNJJ8RmyVE4modEYcnFXEldJlALkEkV0hVRFUTOg8DZeA5IW5vM0W0Fwqtor1hMitNCpzgoQA2gBdaWysTaqSyMV2ohqhIyspGEw4SzWWwKDVahUBh0G8BG1nspYWvU2jbB5XBx3E9W++WSJVB-Gqp1S8NCSP0MmkVBiBNwf2MB1rIRYGDCKBcngi4BzOUd2ZwBs4BwYvh9ZbwHBkvDADrkwJmsI1mJFRhe+LDPZ7AGYnk53fSA84OY4AFAA) ("multi" should be required in some cases?)

Reply:
When we want to provide a multi token we provide `T` for a token that expects `T[]`, for example `NG_VALIDATORS`